### ISSUE TITLE
Closing stdin after all buffered writes are completed?

### ISSUE THREAD

**Issue by KittyMac** [brettwooldridge/NuProcess#14]
Is it possible to tell a NuProcess to close stdin after it finishes writing all of the supplied data asynchronously?  In the example, you close stdin after you successfully read from the stdout of the child process.  In my case, I am calling a tool which expect the parent to close stdin after writing all of the data it wants to send, then it processes, then it write back.

I tried directly writing to process.writeStdin() and by using the onStdinReady() callback.  Calling nuProcess.closeStdin() does not appear to be the way to go, as the docs say it will close immediately.  I do not see a method to check the process of the buffered writes to stdin.

**Comment by hyee** [comment]
I have similar requirement, the difference is that I want to know if the child process is running the command, or has finished the task and is waiting for another input.

**Comment by KittyMac** [comment]
For what its worth, I have successfully hacked around it for my (very limited) case.  In BasePosixProcess, in public boolean writeStdin(int availability), I added the following close:

      if (!userWantsWrite.get()) {
		LibC.close(fd);
         return false;
      }

This gives the behavior of closing std after the first write.  Obviously breaks behavior for anything that needs to write multiple times, but none of my tools care about that.
